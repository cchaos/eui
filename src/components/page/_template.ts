/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

export const TEMPLATES = [
  'default',
  'centeredBody',
  'centeredContent',
  'empty',
] as const;

export type _EuiPageTemplate = {
  /**
   * Choose between 3 types of templates.
   * `default`: Typical layout with nothing centered
   * `centeredBody`: The panelled content is centered
   * `centeredContent`: The content inside the panel is centered
   * `empty`: Removes the panneling of the page content
   */
  template?: typeof TEMPLATES[number];
};
