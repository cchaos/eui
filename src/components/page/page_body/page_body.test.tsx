/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../../test/required_props';

import { EuiPageBody } from './page_body';
import { TEMPLATES } from '../_template';

describe('EuiPageBody', () => {
  describe('template', () => {
    TEMPLATES.forEach((template) => {
      describe(`${template}`, () => {
        test('is rendered', () => {
          const component = render(<EuiPageBody {...requiredProps} />);

          expect(component).toMatchSnapshot();
        });
        it('paddingSize is rendered', () => {
          const component = render(
            <EuiPageBody template={template} paddingSize="none" />
          );

          expect(component).toMatchSnapshot();
        });

        test('is rendered with pageHeader', () => {
          const component = render(
            <EuiPageBody
              template={template}
              pageHeader={{
                title: 'Page title',
                ...requiredProps,
              }}
            />
          );

          expect(component).toMatchSnapshot();
        });

        test('is rendered with pageContentProps', () => {
          const component = render(
            <EuiPageBody template={template} pageContentProps={requiredProps} />
          );

          expect(component).toMatchSnapshot();
        });

        test('is rendered with pageContentBodyProps', () => {
          const component = render(
            <EuiPageBody
              template={template}
              pageContentBodyProps={requiredProps}
            />
          );

          expect(component).toMatchSnapshot();
        });

        test('is rendered with bottomBar', () => {
          const component = render(
            <EuiPageBody template={template} bottomBar="Bottom bar" />
          );

          expect(component).toMatchSnapshot();
        });

        describe('panelled', () => {
          test('can be set to true', () => {
            const component = render(<EuiPageBody panelled={true} />);

            expect(component).toMatchSnapshot();
          });

          test('also accepts panelProps', () => {
            const component = render(
              <EuiPageBody panelled={true} panelProps={{ color: 'subdued' }} />
            );

            expect(component).toMatchSnapshot();
          });
        });
      });
    });
  });

  describe('restrict width', () => {
    test('can be set to a default', () => {
      const component = render(
        <EuiPageBody {...requiredProps} restrictWidth={true} />
      );

      expect(component).toMatchSnapshot();
    });

    test('can be set to a custom number', () => {
      const component = render(
        <EuiPageBody {...requiredProps} restrictWidth={1024} />
      );

      expect(component).toMatchSnapshot();
    });

    test('can be set to a custom value and measurement', () => {
      const component = render(
        <EuiPageBody {...requiredProps} restrictWidth="24rem" />
      );

      expect(component).toMatchSnapshot();
    });
  });
});
