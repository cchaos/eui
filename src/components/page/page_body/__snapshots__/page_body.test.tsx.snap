// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiPageBody restrict width can be set to a custom number 1`] = `
<div
  aria-label="aria-label"
  class="euiPageBody euiPageBody--restrictWidth-custom testClass1 testClass2"
  data-test-subj="test subject string"
  style="max-width:1024px"
/>
`;

exports[`EuiPageBody restrict width can be set to a custom value and measurement 1`] = `
<div
  aria-label="aria-label"
  class="euiPageBody euiPageBody--restrictWidth-custom testClass1 testClass2"
  data-test-subj="test subject string"
  style="max-width:24rem"
/>
`;

exports[`EuiPageBody restrict width can be set to a default 1`] = `
<div
  aria-label="aria-label"
  class="euiPageBody euiPageBody--restrictWidth-default testClass1 testClass2"
  data-test-subj="test subject string"
/>
`;

exports[`EuiPageBody template centeredBody is rendered 1`] = `
<div
  aria-label="aria-label"
  class="euiPageBody testClass1 testClass2"
  data-test-subj="test subject string"
/>
`;

exports[`EuiPageBody template centeredBody is rendered with bottomBar 1`] = `
<div
  class="euiPageBody"
>
  <div
    class="euiPanel euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPageContent euiPageContent--verticalCenter euiPageContent--horizontalCenter"
    role="main"
  >
    <div
      class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiPageContentBody"
    />
  </div>
</div>
`;

exports[`EuiPageBody template centeredBody is rendered with pageContentBodyProps 1`] = `
<div
  class="euiPageBody"
>
  <div
    class="euiPanel euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPageContent euiPageContent--verticalCenter euiPageContent--horizontalCenter"
    role="main"
  >
    <div
      aria-label="aria-label"
      class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiPageContentBody testClass1 testClass2"
      data-test-subj="test subject string"
    />
  </div>
</div>
`;

exports[`EuiPageBody template centeredBody is rendered with pageContentProps 1`] = `
<div
  class="euiPageBody"
>
  <div
    aria-label="aria-label"
    class="euiPanel euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPageContent euiPageContent--verticalCenter euiPageContent--horizontalCenter testClass1 testClass2"
    data-test-subj="test subject string"
    role="main"
  >
    <div
      class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiPageContentBody"
    />
  </div>
</div>
`;

exports[`EuiPageBody template centeredBody is rendered with pageHeader 1`] = `
<div
  class="euiPageBody"
>
  <div
    class="euiPanel euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPageContent euiPageContent--verticalCenter euiPageContent--horizontalCenter"
    role="main"
  >
    <header
      aria-label="aria-label"
      class="euiPageHeader euiPageHeader--responsive euiPageHeader--center testClass1 testClass2"
      data-test-subj="test subject string"
      title="Page title"
    />
    <div
      class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--hasBorder euiPageContentBody"
      style="border-width:1px 0 0 0"
    />
  </div>
</div>
`;

exports[`EuiPageBody template centeredBody paddingSize is rendered 1`] = `
<div
  class="euiPageBody"
>
  <div
    class="euiPanel euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPageContent euiPageContent--verticalCenter euiPageContent--horizontalCenter"
    role="main"
  >
    <div
      class="euiPanel euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiPageContentBody"
    />
  </div>
</div>
`;

exports[`EuiPageBody template centeredBody panelled also accepts panelProps 1`] = `
<div
  class="euiPanel euiPanel--borderRadiusMedium euiPanel--subdued euiPanel--noShadow euiPanel--noBorder euiPageBody"
>
  <div
    class="euiPageBody"
  />
</div>
`;

exports[`EuiPageBody template centeredBody panelled can be set to true 1`] = `
<div
  class="euiPanel euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPageBody"
>
  <div
    class="euiPageBody"
  />
</div>
`;

exports[`EuiPageBody template centeredContent is rendered 1`] = `
<div
  aria-label="aria-label"
  class="euiPageBody testClass1 testClass2"
  data-test-subj="test subject string"
/>
`;

exports[`EuiPageBody template centeredContent is rendered with bottomBar 1`] = `
<div
  class="euiPageBody"
>
  <div
    class="euiPanel euiPanel--paddingLarge euiPanel--borderRadiusNone euiPanel--plain euiPanel--noShadow euiPanel--noBorder euiPageContent euiPageContent--flex"
    role="main"
  >
    <div
      class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--subdued euiPanel--noShadow euiPanel--noBorder euiPageContentBody euiPageContentBody--centeredContent"
    />
  </div>
  <section
    aria-label="Page level controls"
    class="euiBottomBar euiBottomBar--sticky"
    style="left:0;right:0;bottom:0"
  >
    <h2
      class="euiScreenReaderOnly"
    >
      Page level controls
    </h2>
    <div
      class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiPageContentBody"
    >
      Bottom bar
    </div>
  </section>
  <p
    aria-live="assertive"
    class="euiScreenReaderOnly"
  >
    There is a new region landmark with page level controls at the end of the document.
  </p>
</div>
`;

exports[`EuiPageBody template centeredContent is rendered with pageContentBodyProps 1`] = `
<div
  class="euiPageBody"
>
  <div
    class="euiPanel euiPanel--paddingLarge euiPanel--borderRadiusNone euiPanel--plain euiPanel--noShadow euiPanel--noBorder euiPageContent euiPageContent--flex"
    role="main"
  >
    <div
      aria-label="aria-label"
      class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--subdued euiPanel--noShadow euiPanel--noBorder euiPageContentBody euiPageContentBody--centeredContent testClass1 testClass2"
      data-test-subj="test subject string"
    />
  </div>
</div>
`;

exports[`EuiPageBody template centeredContent is rendered with pageContentProps 1`] = `
<div
  class="euiPageBody"
>
  <div
    aria-label="aria-label"
    class="euiPanel euiPanel--paddingLarge euiPanel--borderRadiusNone euiPanel--plain euiPanel--noShadow euiPanel--noBorder euiPageContent euiPageContent--flex testClass1 testClass2"
    data-test-subj="test subject string"
    role="main"
  >
    <div
      class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--subdued euiPanel--noShadow euiPanel--noBorder euiPageContentBody euiPageContentBody--centeredContent"
    />
  </div>
</div>
`;

exports[`EuiPageBody template centeredContent is rendered with pageHeader 1`] = `
<div
  class="euiPageBody"
>
  <header
    aria-label="aria-label"
    class="euiPageHeader euiPageHeader--responsive euiPageHeader--center testClass1 testClass2"
    data-test-subj="test subject string"
    title="Page title"
  />
  <div
    class="euiPanel euiPanel--paddingLarge euiPanel--borderRadiusNone euiPanel--plain euiPanel--noShadow euiPanel--hasBorder euiPageContent euiPageContent--flex"
    role="main"
    style="border-width:1px 0 0 0"
  >
    <div
      class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--subdued euiPanel--noShadow euiPanel--noBorder euiPageContentBody euiPageContentBody--centeredContent"
    />
  </div>
</div>
`;

exports[`EuiPageBody template centeredContent paddingSize is rendered 1`] = `
<div
  class="euiPageBody"
>
  <div
    class="euiPanel euiPanel--paddingLarge euiPanel--borderRadiusNone euiPanel--plain euiPanel--noShadow euiPanel--noBorder euiPageContent euiPageContent--flex"
    role="main"
  >
    <div
      class="euiPanel euiPanel--borderRadiusMedium euiPanel--subdued euiPanel--noShadow euiPanel--noBorder euiPageContentBody euiPageContentBody--centeredContent"
    />
  </div>
</div>
`;

exports[`EuiPageBody template centeredContent panelled also accepts panelProps 1`] = `
<div
  class="euiPanel euiPanel--borderRadiusMedium euiPanel--subdued euiPanel--noShadow euiPanel--noBorder euiPageBody"
>
  <div
    class="euiPageBody"
  />
</div>
`;

exports[`EuiPageBody template centeredContent panelled can be set to true 1`] = `
<div
  class="euiPanel euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPageBody"
>
  <div
    class="euiPageBody"
  />
</div>
`;

exports[`EuiPageBody template default is rendered 1`] = `
<div
  aria-label="aria-label"
  class="euiPageBody testClass1 testClass2"
  data-test-subj="test subject string"
/>
`;

exports[`EuiPageBody template default is rendered with bottomBar 1`] = `
<div
  class="euiPageBody"
>
  <div
    class="euiPanel euiPanel--borderRadiusNone euiPanel--plain euiPanel--noShadow euiPanel--noBorder euiPageContent"
    role="main"
  >
    <div
      class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiPageContentBody"
    />
  </div>
  <section
    aria-label="Page level controls"
    class="euiBottomBar euiBottomBar--sticky"
    style="left:0;right:0;bottom:0"
  >
    <h2
      class="euiScreenReaderOnly"
    >
      Page level controls
    </h2>
    <div
      class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiPageContentBody"
    >
      Bottom bar
    </div>
  </section>
  <p
    aria-live="assertive"
    class="euiScreenReaderOnly"
  >
    There is a new region landmark with page level controls at the end of the document.
  </p>
</div>
`;

exports[`EuiPageBody template default is rendered with pageContentBodyProps 1`] = `
<div
  class="euiPageBody"
>
  <div
    class="euiPanel euiPanel--borderRadiusNone euiPanel--plain euiPanel--noShadow euiPanel--noBorder euiPageContent"
    role="main"
  >
    <div
      aria-label="aria-label"
      class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiPageContentBody testClass1 testClass2"
      data-test-subj="test subject string"
    />
  </div>
</div>
`;

exports[`EuiPageBody template default is rendered with pageContentProps 1`] = `
<div
  class="euiPageBody"
>
  <div
    aria-label="aria-label"
    class="euiPanel euiPanel--borderRadiusNone euiPanel--plain euiPanel--noShadow euiPanel--noBorder euiPageContent testClass1 testClass2"
    data-test-subj="test subject string"
    role="main"
  >
    <div
      class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiPageContentBody"
    />
  </div>
</div>
`;

exports[`EuiPageBody template default is rendered with pageHeader 1`] = `
<div
  class="euiPageBody"
>
  <header
    aria-label="aria-label"
    class="euiPageHeader euiPageHeader--responsive euiPageHeader--center testClass1 testClass2"
    data-test-subj="test subject string"
    title="Page title"
  />
  <div
    class="euiPanel euiPanel--borderRadiusNone euiPanel--plain euiPanel--noShadow euiPanel--hasBorder euiPageContent"
    role="main"
    style="border-width:1px 0 0 0"
  >
    <div
      class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiPageContentBody"
    />
  </div>
</div>
`;

exports[`EuiPageBody template default paddingSize is rendered 1`] = `
<div
  class="euiPageBody"
>
  <div
    class="euiPanel euiPanel--borderRadiusNone euiPanel--plain euiPanel--noShadow euiPanel--noBorder euiPageContent"
    role="main"
  >
    <div
      class="euiPanel euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiPageContentBody"
    />
  </div>
</div>
`;

exports[`EuiPageBody template default panelled also accepts panelProps 1`] = `
<div
  class="euiPanel euiPanel--borderRadiusMedium euiPanel--subdued euiPanel--noShadow euiPanel--noBorder euiPageBody"
>
  <div
    class="euiPageBody"
  />
</div>
`;

exports[`EuiPageBody template default panelled can be set to true 1`] = `
<div
  class="euiPanel euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPageBody"
>
  <div
    class="euiPageBody"
  />
</div>
`;

exports[`EuiPageBody template empty is rendered 1`] = `
<div
  aria-label="aria-label"
  class="euiPageBody testClass1 testClass2"
  data-test-subj="test subject string"
/>
`;

exports[`EuiPageBody template empty is rendered with bottomBar 1`] = `
<div
  class="euiPageBody"
>
  <div
    class="euiPanel euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiPageContent"
    role="main"
  >
    <div
      class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiPageContentBody"
    />
  </div>
  <section
    aria-label="Page level controls"
    class="euiBottomBar euiBottomBar--sticky"
    style="left:0;right:0;bottom:0"
  >
    <h2
      class="euiScreenReaderOnly"
    >
      Page level controls
    </h2>
    <div
      class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiPageContentBody"
    >
      Bottom bar
    </div>
  </section>
  <p
    aria-live="assertive"
    class="euiScreenReaderOnly"
  >
    There is a new region landmark with page level controls at the end of the document.
  </p>
</div>
`;

exports[`EuiPageBody template empty is rendered with pageContentBodyProps 1`] = `
<div
  class="euiPageBody"
>
  <div
    class="euiPanel euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiPageContent"
    role="main"
  >
    <div
      aria-label="aria-label"
      class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiPageContentBody testClass1 testClass2"
      data-test-subj="test subject string"
    />
  </div>
</div>
`;

exports[`EuiPageBody template empty is rendered with pageContentProps 1`] = `
<div
  class="euiPageBody"
>
  <div
    aria-label="aria-label"
    class="euiPanel euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiPageContent testClass1 testClass2"
    data-test-subj="test subject string"
    role="main"
  >
    <div
      class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiPageContentBody"
    />
  </div>
</div>
`;

exports[`EuiPageBody template empty is rendered with pageHeader 1`] = `
<div
  class="euiPageBody"
>
  <header
    aria-label="aria-label"
    class="euiPageHeader euiPageHeader--responsive euiPageHeader--center testClass1 testClass2"
    data-test-subj="test subject string"
    title="Page title"
  />
  <div
    class="euiPanel euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--hasBorder euiPageContent"
    role="main"
    style="border-width:1px 0 0 0"
  >
    <div
      class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiPageContentBody"
    />
  </div>
</div>
`;

exports[`EuiPageBody template empty paddingSize is rendered 1`] = `
<div
  class="euiPageBody"
>
  <div
    class="euiPanel euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiPageContent"
    role="main"
  >
    <div
      class="euiPanel euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiPageContentBody"
    />
  </div>
</div>
`;

exports[`EuiPageBody template empty panelled also accepts panelProps 1`] = `
<div
  class="euiPanel euiPanel--borderRadiusMedium euiPanel--subdued euiPanel--noShadow euiPanel--noBorder euiPageBody"
>
  <div
    class="euiPageBody"
  />
</div>
`;

exports[`EuiPageBody template empty panelled can be set to true 1`] = `
<div
  class="euiPanel euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPageBody"
>
  <div
    class="euiPageBody"
  />
</div>
`;
